src/containers/Dashboard/NodeSlice.ts

```diff
@@ -0,0 +1,56 @@
+import {createSlice, PayloadAction} from "@reduxjs/toolkit";
+import {AppThunk} from "../../app/store";
+import {getNodes, getMinerInfo} from "./DashboardApi";
+import {INodeState, INodeInfoState} from "./NodeInterface";
+
+interface IState {
+    fetchComplete: boolean;
+    nodeList: Array<INodeState>;
+    nodeInfo: INodeInfoState | null;
+}
+const initialState: IState = {
+    fetchComplete: false,
+    nodeList: [],
+    nodeInfo: null,
+};
+
+const nodeSlice = createSlice({
+    name: "node",
+    initialState,
+    reducers: {
+        loading(state: IState): void {
+            state.fetchComplete = true;
+        },
+        storeNodeList(state: IState, action: PayloadAction<Array<INodeState>>): void {
+            action.payload.map(node => state.nodeList.push(node));
+        },
+        storeNodeInfo(state: IState, action: PayloadAction<INodeInfoState>): void {
+            state.nodeInfo = action.payload;
+        },
+    },
+});
+
+export const {loading, storeNodeList, storeNodeInfo} = nodeSlice.actions;
+export default nodeSlice.reducer;
+
+export const getNodeList = (auth: string | null): AppThunk => async dispatch => {
+    try {
+        const nodeListResponse = await getNodes(auth);
+        dispatch(storeNodeList(nodeListResponse.data));
+
+        if (nodeListResponse.data[0] && nodeListResponse.data[0].id) {
+            const nodeInfoResponse = await getMinerInfo(auth, nodeListResponse.data[0].id);
```

Comment:
Functions should have single responsabilities. Meaning, you can't have fetching node list and then getting miner info in the same function. You can leave for now but this logic has to change when resolving #14 anyway

Reply:
Yes now all fetch functions are in the same dispatch getNodeList, I will take 
#14 next and rearrange the logic and implement switching nodes.